GVRf/Framework/jni/objects/mesh.h

```diff
@@ -354,8 +329,11 @@ class Mesh: public HybridObject {
     std::map<int, std::string> attribute_vec4_keys_;
 
     // add vertex array object and VBO
-    GLuint vaoID_;
-    GLuint triangle_vboID_, vert_vboID_, norm_vboID_, tex_vboID_;
+    std::map<Material::ShaderType, GLuint> vaoID_map_;
+    std::map<Material::ShaderType, GLuint> triangle_vboID_map_;
+    std::map<Material::ShaderType, GLuint> vert_vboID_map_;
+    std::map<Material::ShaderType, GLuint> norm_vboID_map_;
+    std::map<Material::ShaderType, GLuint> tex_vboID_map_;
```

Comment:
Just my minor concern here. As far as I can see, actually only we only need map for vaoIDs since only vaoID has getter which requires key. For all other vboID, we just need some kind of container to store them and finally call glDeleteBuffer with all of them.  In std, map is actually a tree(which may differentiate from hash_map), which may be a bit more expensive then something like a dequeue or vector.